### PR TITLE
Some maintenance

### PR DESCRIPTION
**Pull request by ljedrz**
Some maintenance work so as not to conflict too much with the pending PRs:
- depend on cid instead of libipld where only Cid is used
- update the `domain` dep (big `Cargo.lock` wins)

cc #75

### CODE REVIEW

File: src/ipns/dns.rs [hunk 1]
```diff
@@ -57,16 +59,18 @@ impl Future for DnsLinkFuture {
...
+    let qname = Dname::<Bytes>::from_str(&dnslink)?;
     let question = Question::new_in(qname, Rtype::Txt);
-    let query2 = StubResolver::new().query(question).compat();
+    let resolver = StubResolver::new();
+    let query2 = Box::pin(async move { resolver.query(question).await });
 
     Ok(DnsLinkFuture {
-        query: select_ok(vec![query1, query2]),
+        query: select_ok(vec![query1 as FutureAnswer, query2]),
     }
```
Comment (koivunej) [inline src/ipns/dns.rs:74]: did you try either of the test_resolve1 or test_resolve2?

File: tests/wantlist_and_cancellation.rs [hunk 2]
```diff
@@ -2,10 +2,10 @@ use async_std::{
     future::{pending, timeout},
     task,
 };
+use cid::Cid;
 use futures::future::{select, Either, FutureExt};
 use futures::future::{AbortHandle, Abortable};
 use ipfs::Node;
-use libipld::Cid;
```
Comment (koivunej) [inline tests/wantlist_and_cancellation.rs:8]: Great idea; I had been thinking that these were already removed but apparently not!

File: src/ipns/dns.rs (hunk 1 above)
Comment (koivunej) [reply src/ipns/dns.rs:74]: In test_resolve2 we should just expect an IpfsPath and that's it. I did not know of the IpfsPath when writing that FIXME :)

File: tests/wantlist_and_cancellation.rs (hunk 2 above)
Comment (ljedrz) [reply tests/wantlist_and_cancellation.rs:8]: I already did a pass before, but left some stragglers apparently :stuck_out_tongue:

File: src/ipns/dns.rs (hunk 1 above)
Comment (ljedrz) [reply src/ipns/dns.rs:74]: just tried it:
```
test ipns::dns::tests::test_resolve1 ... ok
test ipns::dns::tests::test_resolve2 ... FAILED

failures:

---- ipns::dns::tests::test_resolve2 stdout ----
thread 'ipns::dns::tests::test_resolve2' panicked at 'assertion failed: `(left == right)`
  left: `"/ipfs/bafybeidzfuvdzwkvfv4eonmbtpvkxkwxfxdfowww6wijxoj2chhryf7sc4"`,
 right: `"/ipfs/QmbV3st6TDZVocp4H2f4KE3tvLP1BEpeRHhZyFL9gD4Ut4"`', src/ipns/dns.rs:93:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    ipns::dns::tests::test_resolve2
```

Comment (ljedrz) [reply src/ipns/dns.rs:74]: So I guess resolving works now :stuck_out_tongue:? I can update the path, but should I enable the tests? Probably not, as they could share the same fate as that one `kad` test we disabled today, no?

Comment (koivunej) [reply src/ipns/dns.rs:74]: Great that those work now as well! I this whole thing could be simplified a lot, but probably best to throw this in as is right away. No need to enable that test.. Need to learn if we can have tests enabled only when !ci.

**Pull request merged**